engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -485,11 +488,12 @@ public final ContainerInfo getAppMasterContainerInfo()
         String nodeHttpAddress = nmHost + ":" + nmHttpPort;
         if (allocatedMemoryMB == 0) {
           String url = ConfigUtils.getSchemePrefix(conf) + nodeHttpAddress + "/ws/v1/node/containers/" + ci.id;
-          WebServicesClient webServicesClient = new WebServicesClient();
-          try {
-            String content = webServicesClient.process(url, String.class, new WebServicesClient.GetWebServicesHandler<String>());
-            JSONObject json = new JSONObject(content);
-            int totalMemoryNeededMB = json.getJSONObject("container").getInt("totalMemoryNeededMB");
+          try (YarnClient rmClient = new YarnClientImpl()) {
```

Comment:
Looks like @devtagare is not working on this. I will make the change as I am familiar with this issue. I will add my commit on top so his pull request will also be addressed when merged.